pallets/msa/src/tests.rs

```diff
@@ -2365,3 +2365,175 @@ fn try_mutate_delegation_success() {
 		assert!(Msa::get_delegation(delegator, provider).is_some());
 	});
 }
+
+#[test]
+fn revoke_permissions_for_schema_success() {
+	new_test_ext().execute_with(|| {
+		set_schema_count::<Test>(3);
+
+		let delegator = Delegator(2);
+		let provider = Provider(1);
+		let schema_grants = vec![1];
+
+		assert_ok!(Msa::add_provider(provider, delegator, schema_grants));
+
+		let delegation_relationship = Msa::get_delegation(delegator, provider).unwrap();
+		let mut expected = BoundedBTreeMap::<
+			SchemaId,
+			<Test as frame_system::Config>::BlockNumber,
+			<Test as Config>::MaxSchemaGrantsPerDelegation,
+		>::new();
+
+		expected.try_insert(1, Default::default()).expect("testing expected");
+
+		assert_eq!(delegation_relationship.schema_permissions, expected);
+
+		// Add revoke schema ids
+		let schemas_to_be_revoked = vec![1];
+		let result =
+			Msa::revoke_permissions_for_schemas(delegator, provider, schemas_to_be_revoked);
+
+		assert_ok!(result);
+
+		let delegation_relationship = Msa::get_delegation(delegator, provider).unwrap();
+		let mut expected = BoundedBTreeMap::<
+			SchemaId,
+			<Test as frame_system::Config>::BlockNumber,
+			<Test as Config>::MaxSchemaGrantsPerDelegation,
+		>::new();
+
+		expected.try_insert(1, 1u32.into()).expect("testing expected");
+
+		assert_eq!(delegation_relationship.schema_permissions, expected);
+	});
+}
+
+#[test]
+fn revoke_permissions_for_schemas_errors_when_no_delegation() {
+	new_test_ext().execute_with(|| {
+		let delegator = Delegator(2);
+		let provider = Provider(1);
+		let schema_ids = vec![1, 2];
+		let result = Msa::revoke_permissions_for_schemas(delegator, provider, schema_ids);
+
+		assert_noop!(result, Error::<Test>::DelegationNotFound);
+	});
+}
+
+#[test]
+fn revoke_permissions_for_schemas_errors_when_schema_does_not_exist_in_list_of_schema_grants() {
+	new_test_ext().execute_with(|| {
+		set_schema_count::<Test>(31);
+
+		let delegator = Delegator(2);
+		let provider = Provider(1);
+		let schema_grants = vec![1];
+
+		assert_ok!(Msa::add_provider(provider, delegator, schema_grants));
+
+		let additional_grants = (2..32 as u16).collect::<Vec<_>>();
+		let result = Msa::revoke_permissions_for_schemas(delegator, provider, additional_grants);
+
+		assert_noop!(result, Error::<Test>::SchemaNotGranted);
+
+		let result = Msa::get_delegation(delegator, provider);
+
+		let mut expected = Delegation {
+			revoked_at: 0u32.into(),
+			schema_permissions: BoundedBTreeMap::<
+				SchemaId,
+				<Test as frame_system::Config>::BlockNumber,
+				<Test as Config>::MaxSchemaGrantsPerDelegation,
+			>::new(),
+		};
+
+		expected
+			.schema_permissions
+			.try_insert(1, 0u32.into())
+			.expect("testing expected");
+
+		assert_eq!(result.unwrap(), expected);
+	});
+}
+
+#[test]
+fn revoke_schema_permissions_success() {
+	new_test_ext().execute_with(|| {
+		set_schema_count::<Test>(3);
+
+		let (key_pair, _) = sr25519::Pair::generate();
```

Comment:
You might check and see if `test_create_delegator_msa_with_provider` in mocks will work to save some repeated code.  Even modify it if it would help.